libs/sync/sync.go

```diff
@@ -0,0 +1,33 @@
+package sync
+
+/*
+// For detecting deadlock situations
```

Comment:
Can we re-write this so it's behind a build tag? Then we can easily build binaries with deadlock detection without needing to modify any code!

Reply:
I had problem with `WaitGroup` method. Unfortunately that library doesn't support it. I need to check it more.

Reply:
How about we just write a Makefile target:

```
test_with_deadlock:
    go get github.com/sasha-s/go-deadlock/...
    find . -name "*.go" | xargs -n 1 sed -i '' 's/sync.RWMutex/deadlock.RWMutex/'
    find . -name "*.go" | xargs -n 1 sed -i '' 's/sync.Mutex/deadlock.Mutex/'
    make test
```

?

Reply:
I tried this solution, however I was not successful. Maybe you can do a try as well. 
Also `WaitGroup` is an issue here. Can we not use this struct inside the test files?

Reply:
Can you explain why it is an issue?

Reply:
`‍go-deadlock‍‍` has not such a struct but `sync` has.